src/classes/RD2_CustomFieldMapper_TEST.cls

```diff
@@ -0,0 +1,289 @@
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder().build();
+        Opportunity opp = new TEST_OpportunityBuilder()
+            .withAmount(RD_AMOUNT)
+            .build();
+
+        RD2_CustomFieldMapper fieldMapper = new RD2_CustomFieldMapper();
+        Boolean isChanged = fieldMapper.syncOpportunityFields(opp, rd);
+
+        System.assertEquals(false, isChanged,
+            'The sync should indicate the Opportunity record has not been changed');
+    }
+
+    /***
+     * @description Verifies variables containing Recurring Donation and Opportunity field names
+     * are populated based on the custom field mapping settings
+     */
+    @IsTest
+    private static void shouldPopulateRDandOppFieldNamesWhenCustomMappingIsDefined() {
+        delete [SELECT Id FROM npe03__Custom_Field_Mapping__c];
+
+        insert new List<npe03__Custom_Field_Mapping__c>{
+            buildCustomFieldMapping('npe03__Amount__c', 'Amount'),
+            buildCustomFieldMapping('Name', 'Description')
+        };
+
+        Set<String> expectedRDFieldNames = new Set<String>{
+            'npe03__Amount__c', 'Name'
+        };
+        Set<String> expectedOppFieldNames = new Set<String>{
+            'Amount', 'Description'
+        };
+
+        RD2_CustomFieldMapper fieldMapper = new RD2_CustomFieldMapper();
+
+        System.assertEquals(expectedRDFieldNames, fieldMapper.rdFieldNames,
+            'The mapped field names on Recurring Donation SObject should match');
+        System.assertEquals(expectedOppFieldNames, fieldMapper.oppFieldNames,
+            'The mapped field names on Opportunity SObject should match');
+    }
+
+
+    // Helpers
+    ////////////////////
+
+
+    /**
+     * @description Base common RecurringDonation test structure for all test methods to start from
+     * @return TEST_RecurringDonationBuilder instance
+     */
+    private static TEST_RecurringDonationBuilder getRecurringDonationBuilder() {
+        return getRecurringDonationBuilder(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType));
+    }
+
+    /**
+     * @description Base common RecurringDonation test structure for all test methods to start from
+     * @param contactId Contact Id
+     * @return TEST_RecurringDonationBuilder instance
+     */
+    private static TEST_RecurringDonationBuilder getRecurringDonationBuilder(Id contactId) {
+        return TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+            .withDefaultValues()
+            .withContact(contactId)
+            .withAmount(RD_AMOUNT)
+            .withPaymentMethod(PAYMENT_CHECK)
+            .withDateEstablished(START_DATE)
+            .withStartDate(START_DATE)
+            .withCalculateNextDonationDate();
+    }
+
+    /***
+    * @description Removes current Custom Field Mappings and create a new one
+    * in order to avoid reaching debug log size limit
```

Comment:
How does this avoid reaching the debug log size limit?

Reply:
It is part of the `RD_RecurringDonations_TEST2` comments. I will remove the deletion, not sure why it is there in the first place, but did not want to make tests running into issues if custom settings are not deleted because the approach is used in RD1. Will commit changes to remove deletion in RD2 classes.